Movie Search/Services/NetworkAPI.swift

```diff
@@ -0,0 +1,125 @@
+//
+//  NetworkAPI.swift
+//  Movie Search
+//
+//  Created by Alexey Grabik on 15/04/2020.
+//
+
+import UIKit
+
+protocol NetworkService {
+    func search(for movie: String, completionHandler: @escaping ([MovieModel]?, NetworkServiceError?) -> Void)
```

Comment:
I'd opt for a callback with a `Result<[MovieModel], NetworkingServiceError>` type instead. Same for the UIImage result.